perses/rjmc/topology_proposal.py

```diff
@@ -1109,237 +1106,144 @@ def _is_residue_equal(self, residue, other_residue):
     def _construct_atom_map(self,
                             residue_map,
                             old_topology,
-                            index_to_new_residues,
                             new_topology,
-                            ignore_sidechain_atoms=False):
-        """
-        Construct atom map (key: index to new residue, value: index to old residue) to supply as an argument to the TopologyProposal.
-
+                           extra_sidechain_map=None):
+        """
+        Construct atom map (key: index to atom in new residue, value: index to atom in old residue) to supply as an argument to the TopologyProposal.
+        
+        By default, the atom map:
+        1) Maps all backbone atoms (exception: for GLY, do not map HA2 and HA3)
+        2) Map CBs and HBs only (no other sidechain atoms)
+        
+        If additional sidechains should be mapped, extra_sidechain_map can be supplied to supplement the default map.
+        
         Parameters
         ----------
         residue_map : list(tuples)
             simtk.openmm.app.topology.Residue, str (three letter residue name of new residue)
         old_topology : simtk.openmm.app.Topology
             topology of old system
-        index_to_new_residues : dict
-            key : int (index) , value : str (three letter name of proposed residue)
         new_topology : simtk.openmm.app.Topology
             topology of new system
-        ignore_sidechain_atoms : bool, default False
-            whether to ignore sidechain atoms (CB atoms and beyond) in the atom map
+        extra_sidechain_map : dict, key: int, value: int, default None
+            map of new to old sidechain atom indices to add to the local_atom_map
 
         Returns
         -------
-        adjusted_atom_map : dict, key: int (index
-            new residues have all correct atoms and bonds for desired mutation
-        old_res_to_oemol_map : dict
-            key: int (index);  value: int (index)
-        new_res_to_oemol_map : dict
-            key: int (index);  value: int (index)
-        local_atom_map_stereo_sidechain : dict
-            chirality-corrected map of new_oemol_res to old_oemol_res
-        current_oemol : openeye.oechem.oemol object
-            copy of modified old oemol sidechain
-        proposed_oemol : openeye.oechem.oemol object
-            copy of modified new oemol sidechain
-        old_oemol_res_copy : openeye.oechem.oemol object
-            copy of modified old oemol
-        new_oemol_res_copy : openeye.oechem.oemol object
-            copy of modified new oemol
-
+        local_atom_map : dict, key: int, value int
```

Comment:
i vaguely remember there being a reason we make a deepcopy of the `oemol` to append to the `Topology` object. can we keep that?

Reply:
scratch this. i dont think we need to keep deepcopy anymore.

Reply:
agreed. the deepcopy was necessary before because we were deleting backbone atoms from the original oemol (which would then be inputted to `AtomMapper`). However, we are no longer using the oemol for generating the map.